src/foundation.cs

```diff
@@ -8058,6 +8058,14 @@ interface NSString2 : NSSecureCoding, NSMutableCopying, CKRecordValue
 		[Export ("paragraphRangeForRange:")]
 		NSRange GetParagraphRange (NSRange range);
 
+		[Watch (2,0), TV (9,0), Mac (10,0), iOS (2,0)]
+		[Export("componentsSeparatedByString:")]
+		string[] ComponentsSeparatedByString(string separator);
```

Comment:
ComponentsSeparatedByString -> ComponentsSeparatedBy

Reply:
A few issues

* historically we did not bind API when they were identical to the one already in .net, e.g. `String.Split`. That because of performance (managed-native transition) and to promote re-usable code. Is there a case where this API does something that `String.Split` does not ?

* The name lacks an action/_verb_, something like `SeparateComponents` or just `Split` (close to .net, harder to guess the original selector);

* This is inside `NSString` so, the value I see, would be to split an `NSString` into an array `NSString`, e.g. if it has to be given to another API that accepts `NSString[]`. In that case avoiding the conversion would be beneficial. However having this return `string[]` does look like `String.Split`